test/functional/pdo_sqlsrv/pdo_katmai_special_types.phpt

```diff
@@ -1,150 +1,176 @@
+            }
         }
-
-        var_dump($value);               
-    }    
+        if ($value !== $expOutput[$i]) {
+            echo "Unexpected output retrieved.\n";
+            var_dump($value);
+        }
+    }
+    dropTable($conn, $tableName);
 }
 
-function Katmai_SparseChar($conn)
+function katmaiSparseChar($conn)
 {
-    $tableName = GetTempTableName();
-    
-    $stmt = $conn->exec("CREATE TABLE $tableName (c1 int, c2 char(512) SPARSE NULL, c3 varchar(512) SPARSE NULL, c4 varchar(max) SPARSE NULL, c5 nchar(512) SPARSE NULL, c6 nvarchar(512) SPARSE NULL, c7 nvarchar(max) SPARSE NULL)");   
-    $stmt = null;   
+    $tableName = getTableName();
+
+    // Sparse column set is not supported for Always Encrypted
+    $options = "";
+    if (!isColEncrypted()) {
+        $options = "SPARSE NULL";
+    }
+    $dataTypes = array("c1" => "int",
+                       new ColumnMeta("char(512)", "c2", $options),
+                       new ColumnMeta("char(512)", "c3", $options),
+                       new ColumnMeta("varchar(max)", "c4", $options),
+                       new ColumnMeta("nchar(512)", "c5", $options),
+                       new ColumnMeta("nvarchar(512)", "c6", $options),
+                       new ColumnMeta("nvarchar(max)", "c7", $options));
+    createTable($conn, $tableName, $dataTypes);
 
     $input = "The quick brown fox jumps over the lazy dog";
-    $stmt = $conn->query("INSERT INTO $tableName (c1, c2, c5) VALUES(1, 'The quick brown fox jumps over the lazy dog', 'The quick brown fox jumps over the lazy dog')");  
-    $stmt = null;   
-    $stmt = $conn->query("INSERT INTO $tableName (c1, c3, c6) VALUES(2, 'The quick brown fox jumps over the lazy dog', 'The quick brown fox jumps over the lazy dog')");  
-    $stmt = null;   
-    $stmt = $conn->query("INSERT INTO $tableName (c1, c4, c7) VALUES(3, 'The quick brown fox jumps over the lazy dog', 'The quick brown fox jumps over the lazy dog')");  
-    $stmt = null;   
- 
-    echo "\nComparing results of Katmai sparse fields\n";
-    $stmt = $conn->query("SELECT * FROM $tableName");   
-   
-    while ($row = $stmt->fetch(PDO::FETCH_NUM))
-    {
+    insertRow($conn, $tableName, array("c1" => 1, "c2" => $input, "c5" => $input));
+    insertRow($conn, $tableName, array("c1" => 2, "c3" => $input, "c6" => $input));
+    insertRow($conn, $tableName, array("c1" => 3, "c4" => $input, "c7" => $input));
+
+    echo "Comparing results of Katmai sparse fields\n";
+    $stmt = $conn->query("SELECT * FROM $tableName");
+
+    while ($row = $stmt->fetch(PDO::FETCH_NUM)) {
         $fld1 = $row[0];
         $fld2 = $fld1 + 3;
-        
+
         $value1 = $row[$fld1];
         $value2 = $row[$fld2];
 
-        if ($input !== trim($value1))
-        {
+        if ($input !== trim($value1)) {
             echo "The value is unexpected!\n";
         }
-        if ($value1 !== $value2)
-        {
+        if (trim($value1) !== trim($value2)) {
```

Comment:
why trimming $value2 (not in the original test)
please add a comment to explain